src/items/functions.md

```diff
@@ -214,12 +214,56 @@ extern "C" fn new_i32() -> i32 { 0 }
 let fptr: extern "C" fn() -> i32 = new_i32;
 ```
 
-Functions with an ABI that differs from `"Rust"` do not support unwinding in the
-exact same way that Rust does. Therefore, unwinding past the end of functions
-with such ABIs causes the process to abort.
-
-> **Note**: The LLVM backend of the `rustc` implementation
-aborts the process by executing an illegal instruction.
+### Unwinding
+
+Most ABI strings come in two variants, one with an `-unwind` suffix and one without.
+The `Rust` ABI always permits unwinding, so there is no `Rust-unwind` ABI.
+
+The choice of ABI, together with the runtime [panic mode][panic-modes],
+determines the behavior when unwinding out of a function.
+
+The table below indicates the behavior of an unwinding operation reaching each
+type of ABI boundary (function declaration or definition using the
+corresponding ABI string). Note that the Rust runtime is not affected by, and
+cannot have an effect on, any unwinding that occurs entirely within another
+language's runtime, that is, unwinds that are thrown and caught without
+reaching a Rust ABI boundary.
+
+The `panic`-unwind column refers to [panicking] via the `panic!` macro and
+similar standard library mechanisms, as well as to any other Rust operations
+that cause a panic, such as out-of-bounds array indexing or integer overflow.
+
+The "unwinding" ABI category refers to `"Rust"` (the implicit ABI of Rust
+functions not marked `extern`), `"C-unwind"`, and any other ABI with `-unwind`
+in its name. The "non-unwinding" ABI category refers to all other ABI strings,
+including `"C"` and `"stdcall"`.
+
+Native unwinding is defined per-target. On targets that support throwing and
+catching C++ exceptions, it refers to the mechanism used to implement this
+feature. Some platforms implement a form of unwinding referred to as ["forced
+unwinding"][forced-unwinding]; `longjmp` on Windows and `pthread_exit` in
+`glibc` are implemented this way. Forced unwinding is explicitly excluded
+from the "Native unwind" column in the table.
+
+| panic runtime  | ABI           | `panic`-unwind                        | Native unwind (unforced) |
+| -------------- | ------------  | ------------------------------------- | -----------------------  |
+| `panic=unwind` | unwinding     | unwind                                | unwind                   |
+| `panic=unwind` | non-unwinding | abort (see note below)                | [undefined behavior]     |
+| `panic=abort`  | unwinding     | `panic` aborts without unwinding      | abort                    |
+| `panic=abort`  | non-unwinding | `panic` aborts without unwinding      | [undefined behavior]     |
+
+> Note: With `panic=unwind`, when a `panic` is turned into an abort by a
+> non-unwinding ABI boundary, either no destructors (`Drop` calls) will run, or
+> all destructors up until the ABI boundary will run.
```

Comment:
This note (and the similar one in `src/destructors.md`) address the uncertainty I had in a previous review about what code can potentially run before an abort, thankyou.

Reply:
You're welcome, but be careful, because the behavior in 1.81 _is different_ from what we intend to ship in 1.82. When you asked before, this had been left intentionally undefined, but the lang team later [signed off on](https://github.com/rust-lang/rust/pull/129582#issuecomment-2313670698) the guarantee given in the text here. I had forgotten that the linked PR didn't actually make it into 1.81, though.

Reply:
In that case it would be useful to document that this guarantee only applies from 1.82.

Reply:
All guarantees only apply for the version of the compiler that the documentation ships with. It would be useless clutter to repeat this everywhere. We don't systematically track "when was this feature introduced" in the reference, and this already-long-overdue PR is the wrong place to rehash that choice.